### PR TITLE
Dagster: rename catalog to registry

### LINKED ISSUE #25069: Update catalog naming to be registry
## Overview
We need to start calling our catalogs registry instead.

This ticket is concerned with updating the new dagster system to output the appropriate file paths and use the correct naming in code

## What needs to be updated
1. The names of our `catalog` files and functions in dagster to `registry`
1. The output of `oss_catalog.json` should now be `oss_registry.json`
1. The output of `cloud_catalog.json` should now be `cloud_registry.json`
1. registry files should be output to `registry/` instead of `catalog/`
1. The catalog field in the metadata file should now be registry

## What is out of scope
1. Updating the platform to use the new registry files
1. Updating the platform to use the new registry file names

### PR DESCRIPTION
**Pull request by bnchrch**
## What
1. Renames catalog to registry in Orchestrator
2. Renames catalog to registry in metadata definitions and files
3. Adds a github PAT to the github client to overcome a rate limit
4. Fixes the sensor jobs to materialize parents as well

closes #25069

### CODE REVIEW

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV1.py [hunk 1]
```diff
@@ -15,24 +11,24 @@
 
 
 class ConnectorType(Enum):
-    destination = "destination"
-    source = "source"
+    destination = 'destination'
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV1.py:14]: We generally use double quotes for python strings.

File: airbyte-ci/connectors/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml:1]: I think you should rename these invalid fixture files too.

File: airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/config.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CATALOG_FOLDER = "catalogs"
+REGISTRY_FOLDER = "registry"
```
Comment (alafanechere) [inline airbyte-ci/connectors/metadata_service/orchestrator/orchestrator/config.py:1]:
```suggestion
REGISTRIES_FOLDER = "registries"
```

**Review by alafanechere — CHANGES_REQUESTED** [review]
I think you forgot one renaming: invalid fixture files.
I also don't get why double quotes turned to singles quotes on the generated models.

File: airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV1.py (hunk 1 above)
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/metadata_service/models/generated/ConnectorMetadataDefinitionV1.py:14]: Weird, sorry this was a generated file. Not sure why it changed. Ill look into it

File: airbyte-ci/connectors/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml
Comment (bnchrch) [reply airbyte-ci/connectors/metadata_service/lib/tests/fixtures/invalid/metadata_catalog_no_id_override.yaml:1]: Good call

**Review by alafanechere — APPROVED** [review]

**Pull request merged**